### DISCUSSION TITLE
data-engineering/2024/10/25/duckdb/

### DISCUSSION THREAD

**Discussion by giscus** [zzsza/blog-comments#157] [Comments]
# data-engineering/2024/10/25/duckdb/

이 글은 DuckDB 사용법을 작성한 글입니다 예상 독자 DuckDB가 궁금하신 분 데이터 분석가 : Pandas가 느리다고 생각해서 다른 대안을 찾고 있는 분 데이터 엔지니어 : 데이터 엔지니어링의 Transform 영역에서 활용할 수 있는 수단을 찾고 있는 분. 작은 규모의 회사에서 데이터 웨어하우스를 고민하고 있는 분 키워드 : Duckdb 사용법, Duckdb 설치,...

https://zzsza.github.io/data-engineering/2024/10/25/duckdb/

**Comment by pyeonggwone** [comment]
MVCC 는 ACID 보장이 아니라 말 그대로 데이터의 버전관리만 하는거 아닌가요?
애초에 모든DB 는 ACID 를 지원하는거지 보장하는 개념이 아닙니다
ACID 는 개발자가 쿼리를 작성할때 이를 인지하고 작성해야합니다.

**Reply by zzsza** [reply]
안녕하세요! 데엔방에서 답변드린 내용 옮겨보아요. 말씀해주셔서 감사합니다!

이 부분 제가 너무 짧게 작성했는데, 다시 보니 오해를 할 부분이 있네요. (자체 커스텀한이란 내용이 빠지니 일반적인 MVCC에 대한 이야기 같았네요)
DuckDB는 **자체 커스텀한** MVCC 구현을 통해 ACID를 보장했다고 해요. 커스텀한 방법은 HyPer의 MVCC를 사용했는데, 이 방법은 inplace 업데이트, undo 버퍼로 이전의 상태 관리, OLAP/OLTP 둘 다를 지원하는 워크로드에 최적화한다고 해요. 

자세한 내용은 https://db.in.tum.de/~muehlbau/papers/mvcc.pdf 이 PDF에 자세히 확인할 수 있어요 :)

**Comment by sabper** [comment]
작성해주신 글을 바탕으로 제게 주어진 상황에서 어떻게 활용해볼 수 있을까 고민 하다가 활용해보고 관련 글까지 작성해보았습니다.
좋은 글 써주셔서 감사합니다. 덕분에 큰 도움이 되었습니다.

**Comment by studB** [comment]
많이 배워가게 된 글입니다. 정리 감사합니다.
